lib/osf-components/addon/components/files/manager/component.ts

```diff
@@ -198,6 +200,28 @@ export default class FilesManagerComponent extends Component {
         }
     });
 
+    @enqueueTask({ withTestWaiter: true })
+    deleteFileTask = task(function *(this: FilesManagerComponent, file: File) {
+        try {
+            yield file.delete();
+
+            if (this.onDeleteFile) {
+                this.onDeleteFile(
+                    file,
+                    { callback: file.isFolder ? this.goToParentFolder.bind(this, file) : null },
+                );
+            }
```

Comment:
Out of curiosity, why is this necessary? Rather than, say, just calling `goToParentFolder()` if this is a folder.

Reply:
Can you delete a folder that you're not currently in? What happens then? It seems like this will always bump you up one level regardless of whether that's appropriate or not.

Reply:
Currently, we only support deleting a folder you're currently in. The idea is that it's safer for the user to know the contents of the folder before deleting it. Regarding `goToParentFolder()`, the idea is to keep the files widget generic so it's easier to extend in the future. and because `goToParentFolder` is the most logical thing to do after folder deletion it's passed in as a callback to the consumer of the `<FilesWidget />`.